server/src/test/java/org/opensearch/index/codec/CodecTests.java

```diff
@@ -96,20 +98,23 @@ public void testZlib() throws Exception {
     }
 
     public void testZstd() throws Exception {
+        FeatureFlagSetter.set(FeatureFlags.ZSTD_COMPRESSION);
```

Comment:
Instead of having to explicitly set these in every test method maybe create an `@BeforeClass` method 

```java
@BeforeClass
public static void enableZstd() {
    FeatureFlagSetter.set(FeatureFlags.ZSTD_COMPRESSION);
}
```

Reply:
I wanted to avoid this for tests which are not related to ZSTD, I'll refactor your suggestion

Reply:
The protected method `featureFlagSettings()` is intended to be overridden for use cases like this. I think it is fine to set the flag for all tests in this class, but agree that we should not be doing it by default in the base `OpenSearch[Integ]TestCase`